barretenberg/cpp/CMakeLists.txt

```diff
@@ -123,34 +123,8 @@ if(COVERAGE)
         message(FATAL_ERROR "Creating coverage is only available for clang")
     endif()
 
-    # Get major clang version
```

Comment:
test_coverage.sh is simpler than what we were doing here
we dont need to disable optimization